### PR TITLE
Add outcome_override field

### PR DESCRIPTION
**Pull request by schenedx**
Outcome override field is needed to create outcome description on course run level
EDUCATOR-1694

@clintonb Please review

### CODE REVIEW

File: course_discovery/apps/course_metadata/migrations/0066_auto_20171107_1707.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        ('course_metadata', '0065_program_total_hours_of_effort'),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='courserun',
+            name='outcome_override',
+            field=models.TextField(blank=True, default=None, help_text="What You Will Learn description for this particular course run. Leave this value blank to default to the parent course's Outcome attribute.", null=True),
+        ),
+        migrations.AlterField(
```
Comment (clintonb) [inline course_discovery/apps/course_metadata/migrations/0066_auto_20171107_1707.py]: What is this?

File: course_discovery/apps/course_metadata/tests/factories.py [hunk 2]
```diff
@@ -117,6 +117,7 @@ class CourseRunFactory(factory.DjangoModelFactory):
     hidden = False
     weeks_to_complete = FuzzyInteger(1)
     license = 'all-rights-reserved'
+    outcome_override = None
```
Comment (clintonb) [inline course_discovery/apps/course_metadata/tests/factories.py]: That's the default value. If you aren't going to use a faker, remove this.

File: course_discovery/apps/course_metadata/models.py [hunk 3]
```diff
@@ -598,6 +603,16 @@ def full_description(self, value):
         self.full_description_override = value
 
     @property
+    def outcome(self):
+        return self.outcome_override or self.course.outcome
+
+    @outcome.setter
```
Comment (clintonb) [inline course_discovery/apps/course_metadata/models.py:609]: Why use a setter rather than have the data loader set the override field?

Comment (schenedx) [reply course_discovery/apps/course_metadata/models.py:609]: I am trying to follow the same pattern as we have in "full_description" or "short_description".
Those uses setters.  Why are those coded that way?

File: course_discovery/apps/course_metadata/tests/factories.py (hunk 2 above)
Comment (schenedx) [reply course_discovery/apps/course_metadata/tests/factories.py]: That's fair.

File: course_discovery/apps/course_metadata/migrations/0066_auto_20171107_1707.py (hunk 1 above)
Comment (schenedx) [reply course_discovery/apps/course_metadata/migrations/0066_auto_20171107_1707.py]: Ha.. I didn't notice this existed.
@asadazam93 made this change in https://github.com/edx/course-discovery/pull/1203, and he never did a `makemigrations`. I will remove this from my migrations, and create a new PR just for this migration.

File: course_discovery/apps/course_metadata/models.py (hunk 3 above)
Comment (clintonb) [reply course_discovery/apps/course_metadata/models.py:609]: No clue.

**Comment by schenedx** [comment]
@mikedikan Please help me review this?

**Review by clintonb — APPROVED** [review]

File: course_discovery/apps/course_metadata/models.py [hunk 4]
```diff
@@ -457,6 +457,11 @@ class CourseRun(TimeStampedModel):
     reporting_type = models.CharField(max_length=255, choices=ReportingType.choices, default=ReportingType.mooc)
     eligible_for_financial_aid = models.BooleanField(default=True)
     license = models.CharField(max_length=255, blank=True, db_index=True)
+    outcome_override = models.TextField(
+        default=None, blank=True, null=True,
+        help_text=_(
+            "What You Will Learn description for this particular course run. Leave this value blank to default "
```
Comment (mikedikan) [inline course_discovery/apps/course_metadata/models.py]: i think for clarity, might make sense to change to "'What You Will Learn' description..... (note the single quotes)

**Review by mikedikan — APPROVED** [review]

**Pull request merged**